docs/ops/generation/Multinomial_13.md

```diff
@@ -0,0 +1,181 @@
+* Therefore, in CDF, for every class starting with idx *1* subtract the probability of class *1* = 0.5
+* CDF = ``[0.1, 0.6 - 0.5, 1.0 - 0.5]`` = ``[0.1, 0.1, 0.5]``
+* Normalize CDF by dividing by last value: CDF = ``[0.2, 0.2, 1.0]``
+* Take the next randomly generated float, here 0.6, and repeat until all random samples have assigned classes
+* Assigned classes = ``[1, 2]``
+
+
+**Attributes**:
+
+* ``output_type``
+
+  * **Description**: the type of the output. Determines generation algorithm and affects resulting values. Output numbers generated for different values of *output_type* may not be equal.
+  * **Range of values**: "i4", "i8", "i16", "i32", "i64", "f32", "u4", "u8", "u16", u32", "u64".
+  * **Type**: string
+  * **Required**: *Yes*
+
+* ``with_replacement``
+
+  * **Description**: controls whether to sample with replacement (classes can be sampled multiple times).
+  * **Range of values**: `true`, `false`
+      * ``true`` - class indices can be sampled multiple times.
+      * ``false`` - class indices will not repeat in the output and the size of *input*'s ``class_size`` dimension is required to be larger or equal to *num_samples* value. Might affect performance.
+  * **Type**: `bool`
+  * **Required**: *Yes*
+
+* ``log_probs``
+
+  * **Description**: allows to control whether *inputs* should be treated as probabilities or unnormalized log probabilities.
+  * **Range of values**: `true`, `false`
+      * ``true`` - set values in *inputs* are unnormalized log probabilities that can be any real number.
+      * ``false`` - probabilities in *inputs* are expected to be non-negative, finite and have a non-zero-sum.
+  * **Type**: `bool`
+  * **Required**: *Yes*
+
+* ``global_seed``
+
+  * **Description**: global seed value.
+  * **Range of values**: non-negative integers
+  * **Type**: `unsigned int 64-bit`
+  * **Default value**: 0
+  * **Required**: *Yes*
+
+* ``op_seed``
+
+  * **Description**: operational seed value.
+  * **Range of values**: non-negative integers
+  * **Type**: `unsigned int 64-bit`
+  * **Default value**: 0
+  * **Required**: *Yes*
+
+**Inputs**:
+
+*   **1**: ``input`` - 1D or 2D tensor of type `T_IN` and shape `[class_size]` or `[batch_size, class_size]` with probabilities. Allowed values depend on the *log_probs* attribute. The values are internally normalized to have values in the range of `[0, 1]` with the sum of all probabilities in the given batch equal to 1. **Required.**
+
+*   **2**: ``num_samples`` - scalar or 1D tensor with 1 element of type `T_SAMPLES` specifying the number of samples to draw from Multinomial distribution. **Required.**
+
+**Outputs**:
+
+* **1**:  ``output`` A tensor with type specified by the attribute *output_type* and shape depending on the rank of *input*, either ``[num_samples]`` for one-dimensional *input* or ``[batch_size, num_samples]`` for the two-dimensional one.
+
+**Types**
+
+* **T_IN**: 32-bit floating point numbers.
```

Comment:
Is there specific reason for 32-bit float restriction for the probs input type?
If not, let's keep it as usual:
```suggestion
* **T_IN**: any supported floating-point type.
```
As the supported types are often plugin or implementation dependent, the specification can be more generic.
Also for example ONNX Multinomial supports `tensor(double), tensor(float), tensor(float16)` as the first input type, so with fp32 restriction additional Convert would be needed to align them.

Reply:
I changed it to `f32` because of the PR Paweł mentioned https://github.com/openvinotoolkit/openvino/pull/19830
If I understand correctly, it is supposed to remove support for floating-point type inputs `f64`, `f16` and `bf16`, so the only one remaining would be `f32`

Reply:
The mention PR removes from operators evaluate these precisions as it is used for shape_inference, constant folding or sometimes as fallback path for data inference.
The operator can still provide support for these precisions but maybe is not required to provide them into op's evaluate (plugin can still implement it).

This removal is more like trigger to double check if these precision should be supported and if yes it can be still added in specification.